### PR TITLE
handle default auth type

### PR DESCRIPTION
**Pull request by alanrenmsft**
this PR fixes https://github.com/microsoft/azuredatastudio-mysql/issues/46
I wanted to transfer it to ads repo but couldn't.

the issue happened because we are assigning the AzureMFA as the default auth type and it fails the check for AzureMFA authentication type. 

fix: do not set authentication type if the auth type option is not specified by the connection provider.

### CODE REVIEW

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts [hunk 1]
```diff
@@ -797,8 +797,9 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return defaultProvider && this._providers.has(defaultProvider) ? defaultProvider : undefined;
 	}
 
-	public getDefaultAuthenticationTypeId(): string {
-		let defaultAuthenticationType = WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultAuthenticationType);
+	public getDefaultAuthenticationTypeId(providerName: string): string {
+		const authOption = this._capabilitiesService.getCapabilities(providerName)?.connection.connectionOptions?.find(option => option.specialValueType === ConnectionOptionSpecialType.authType);
+		const defaultAuthenticationType = authOption ? WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultAuthenticationType) : undefined;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionManagementService.ts:802]: Shouldn't we also be checking to see if the auth type is actually supported by the provider? Right now if they only support basic as an auth type for example it'll still pull whatever value is configured there.

And if we're talking about further improvements - it also seems like this should be something set per-provider. Having a single default auth type for everyone isn't going to work long term either.

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts [hunk 2]
```diff
@@ -797,8 +797,9 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return defaultProvider && this._providers.has(defaultProvider) ? defaultProvider : undefined;
 	}
 
-	public getDefaultAuthenticationTypeId(): string {
-		let defaultAuthenticationType = WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultAuthenticationType);
+	public getDefaultAuthenticationTypeId(providerName: string): string {
+		const authOption = this._capabilitiesService.getCapabilities(providerName)?.connection.connectionOptions?.find(option => option.specialValueType === ConnectionOptionSpecialType.authType);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/connectionManagementService.ts:801]: Comment explaining this

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/connection/browser/connectionManagementService.ts:802]: I have made an issue for this: https://github.com/microsoft/azuredatastudio/issues/20573

File: src/sql/workbench/services/connection/browser/connectionManagementService.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/workbench/services/connection/browser/connectionManagementService.ts:803]: done.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**